contracts/Issuehunter.sol

```diff
@@ -129,16 +135,24 @@ contract Issuehunter {
 
     // Submit a new patch.
     //
-    // TODO: the sender must pay a fee that will be used by the patch verifier
-    // to send the transaction to verify the patch.
-    function submitPatch(bytes32 issueId, bytes32 ref) public {
+    // This method is defined as payable because the sender must pay a patch
+    // verification fee that will be used by the patch verifier, after the patch
+    // has been verified, to inform the contract of the successful verification.
+    function submitPatch(bytes32 issueId, bytes32 ref) public payable {
         // Require that a campaign exists
         require(campaigns[issueId].createdBy != 0);
         // Fail if sender already submitted the same patch
         require(campaigns[issueId].patches[msg.sender] == 0 || campaigns[issueId].patches[msg.sender] != ref);
 
         // TODO: require that a campaign hasn't any verified patch
 
+        // Calculate fee amount based on the current transaction's gas price
+        uint feeAmount = _patchVerificationFee(tx.gasprice);
+        // Fail if the transaction value is less than the verification fee
+        require(msg.value >= feeAmount);
```

Comment:
Is value already checked to be also greater than 0? Maybe it is impossible to create a msg with a negative value... Just wondering.

Reply:
`feeAmount >= 0` by definition, so the check `msg.value >= feeAmount` implies that `msg.value >= 0`, correct? Moreover what would it mean to send negative transaction? Does it mean that you can steal money from other accounts by crafting a transaction that include a negative value? I don't think it should be possible.